test/hoctests/tests/test_many_ions.py

```diff
@@ -0,0 +1,95 @@
+# Test when large number of ion mechanisms exist.
+
+# Strategy is to create a large number of ion mechanisms before loading
+# the neurondemo mod file shared library which will create the ca ion
+# along with several mechanisms that write to cai.
+
+from neuron import h, load_mechanisms
+from platform import machine
+import sys
+import io
+
+
+# return True if name exists in h
+def exists(name):
+    try:
+        exec("h.%s" % name)
+    except:
+        return False
+    return True
+
+
+# use up a large number of mechanism indices for ions. And want to test beyond
+# 1000 which requires change to max_ions in nrn/src/nrnoc/eion.cpp
+nion = 250
+ion_indices = [int(h.ion_register("ca%d" % i, 2)) for i in range(1, nion + 1)]
+# gain some confidence they exist
+assert exists("ca%d_ion" % nion)
+assert (ion_indices[-1] - ion_indices[0]) == (nion - 1)
+mt = h.MechanismType(0)
+assert mt.count() > nion
+
+# this test depends on the ca ion not existing at this point
+assert exists("ca_ion") is False
+
+# load neurondemo mod files. That will create ca_ion and provide two
+# mod files that write cai
+# Following Aborts prior to PR#3055 with
+# eion.cpp:431: void nrn_check_conc_write(Prop*, Prop*, int): Assertion `k < sizeof(long) * 8' failed.
+nrnmechlibpath = "%s/demo/release" % h.neuronhome()
+print(nrnmechlibpath)
+assert load_mechanisms(nrnmechlibpath)
```

Comment:
These lines seem to be what's causing the issue with Coverage. The error message we see is related to generated converage reports for the "same" object file twice.

I suspect that somehow when we run the demo as a test, this causes coverage to be collected for the same object file twice.

I tried copying the two modfiles `cabpmp/cadifpmp` and `capmpr` to the parent directory. The this tests crashes slightly differently, but `test_neurondemo` works again.

Reply:
The test relies on deferred loading of libnrnmech.so (which creates the ```ca_ion``` and two mechanisms that write to cai) until after a large number of other ions are created via the interpreter. This causes ```ca_ion``` and the two mechanisms to have large mechanism type values so that we can test what happens when ```ca_ion``` has a mechanism type index > 32 (and eventually > 1000).  Since the neurondemo libnrnmech.so is a well-known model library that is always findable, I thought I'd use that (just had to add another (unused in the demo per se) mechanism that writes to cai) to test the warning for when two mechanisms exist in the same compartment that write to cai.

Reply:
I should mention that share/examples/nrniv/nmodl/capmp.mod is copied to share/demo/release/capmpr.mod
so the filenames are different as well as the SUFFIX